DXRModules/DeusEx/Classes/DXREventsBase.uc

```diff
@@ -1160,6 +1176,11 @@ function _MarkBingo(coerce string eventname)
     }
 
     data = class'PlayerDataItem'.static.GiveItem(player());
+    if (data.IsBingoFailed(eventname)) {
```

Comment:
why don't we just move this logic into the `data.IncrementBingoProgress(eventname)` function, so that it can use a single for loop instead of doing 2 separate for loops

Reply:
My thinking was that `PlayerDataItem` should respect whatever it's told to do, and that this sort of logic should be left to higher level interface functions.